### ISSUE TITLE
Add -l parameter to install/config scripts to overwrite the log directory

### ISSUE THREAD

**Issue by stevenhorsman** [zowe/zowe-install-packaging#1154]
We currently have a few different scripts that each log during running:

install, certificate setup, proclib copying...
these each write logs in a slightly different place (or different files in the home directory), making management annoying.

For the pipeline (at least) adding a -l (lowercase L) parameter specifying the log directory for each step might be the best solution.

Needs to update:
- [x] zowe-install.sh - Note this file is moved into the runtime and deleted at the end
- ~zowe-configure-instance.sh~ this defaults to the instance directory so we can access these in service
- [x] zowe-setup-certificates.sh

Blocked by #1156 
- ~[ ] zowe-copy-to-JES.sh~
- ~[ ] zowe-install-proc.sh~
- ~[ ] zowe-install-xmem.sh~

**Question**
Given that the current behaviour of zowe-install.sh is, on completion to move the log into the runtime install (so that we can pick it up there for service purposes), how do we want to handle -l?
Do we:
1. Create the temp log location in -l then move to runtime
2. Create the log location in -l then copy to runtime, so it is in both
3. Create the log location in -l and don't move at the end (this might cause issues with the service process)

**Comment by stevenhorsman** [comment]
@OnnoVdT - I'm guessing that you would vote for option 2 or in 3 in the question above for the smpe-install?
@jackjia-ibm - I think I raised the original issues after talking to you about log management on the pipeline. I'm guessing that the zowe-install.sh log output isn't a problem for you, but thought I'd check if you had any strong feeling about the question as well?

**Comment by OnnoVdT** [comment]
As mentioned in #801, smpe-install.sh currently removes an log directory it finds in the root of the installed product before doing the SMPE specific repackaging, as people installing with SMPE have no use for our install log, and it would be confusing for support if it was still there. 
It would be nice if we don't have to do the cleanup (less chance for mishaps, for example when the log dir is renamed), so I'm in favor of not placing the log in runtime during or after install, but only if instructed to do so, default behavior should be to have the log in the runtime for pax-isntall support purposes.
Besides the location of the log file as discussed here, it would also be good if smpe could control the name of the log file itself, but that is covered in #801. (If smpe controls the name it can also manage cleanup of older logs, as this data survives builds via a trip through artifactory.)
Keep me in the loop on changes done for this issue, and I'll make sure the smpe side of the story follows suit, keeping the discussions/work in #801 in mind.

**Comment by jackjia-ibm** [comment]
Thanks @stevenhorsman. The `zowe-install.sh` log is not an issue for now, it's pretty consistent. I'm ok with either way. But I didn't get how it could cause issues with the service process if we choose option 3. Is there another process after installation will rely on the logs? If not, I would prefer option 3.

Besides the logs shown above, we also have logs in these format:

- ~/zowe-copy-to-JES.sh-<timestamp>.log
- ~/zowe-install-proc.sh-<timestamp>.log
- ~/zowe-install-xmem.sh-<timestamp>.log

If we can put those logs also setup-certificates log all into one place will be ideal for me. To me, those are also part of installation log and I think the user who perform those steps should have write permission to the log folder where zowe-install.sh log locates. On Linux, `/var/log` is a pretty common place to put logs, which can be default value, but seems not same on z/OS.

Comparing to #801, if we can keep it consistent, both use `-l` or `-f`, will be better.

**Comment by stevenhorsman** [comment]
My concern with option 3 is that if there was a partial failure during install (a chgrp, or component failure) then we'd want to know when servicing the product, and if it's in a random place we couldn't 'gather' that info

**Comment by jackjia-ibm** [comment]
Thanks Steve, I see what you mean on the service process and it makes sense. I was thinking if some zowe software program/service are relying on the logs.

**Comment by stevenhorsman** [comment]
FYI @OnnoVdT @jackjia-ibm. After speaking to Joe we've decided that splitting into the log without doing updating the default log directory is probable going to cause more mess. We've updated the parent epic we've a combined proposal which is closest to option 3 above and will apply to the zowe-install, zowe-setup-certificates and 3 dataset scripts at the same time in the same way. Take a look at the proposal in https://github.com/zowe/zowe-install-packaging/issues/956 and I'll probably close this issue out.
In order to help get towards the requirements that Onno wants for the smpe-build in particular we are proposing not moving the final log to runtime and outputting the log file in stdout, so either the output could be parsed to find the file, or the log-directory location ls'd to find the latest file.
